pymongo_auth_aws/auth.py

```diff
@@ -45,6 +47,26 @@ def _aws_temp_credentials():
     if access_key and secret_key:
         return AwsCredential(
             access_key, secret_key, os.environ.get('AWS_SESSION_TOKEN'))
+    # Check if environment variables exposed by IAM Roles for Service Accounts (IRSA) are present.
+    # See https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html for details.
+    irsa_web_id_file = os.getenv('AWS_WEB_IDENTITY_TOKEN_FILE')
+    irsa_role_arn = os.getenv('AWS_ROLE_ARN')
+    if irsa_web_id_file and irsa_role_arn:
+        try:
+            with open(irsa_web_id_file) as f:
+                irsa_web_id_token = f.read()
+        # Check for errors raised by `open` from older Python versions.
+        except (OSError, IOError, InterruptedError) as exc:
```

Comment:
InterruptedError doesn't exist in Python 2 and is a subclass of OSError in Python 3 so we should remove it.

Reply:
Removed